### PR TITLE
first stab at type annotations

### PR DESCRIPTION
**Pull request by marcmengel**
Type annotations on functions/args throughout.

### CODE REVIEW

File: bin/condor_submit [hunk 1]
```diff
@@ -50,7 +50,7 @@ COLLECTOR_HOST = htcondor.param.get("COLLECTOR_HOST", "gpcollector03.fnal.gov")
 class StoreGroupinEnvironment(argparse.Action):
     """Action to store the given group in the GROUP environment variable"""
 
-    def __call__(self, parser, namespace, values, option_string=None):
+    def __call__(self, parser:argparse.ArgumentParser, namespace:Any, values:Dict[str,str], option_string=None:Union[None,str]):
```
Comment (shreyb) [inline bin/condor_submit:53]: Any reason the `namespace` is type `Any` when we're using this with `argparse`, so the expected type is `argparse.Namespace`?

File: bin/jobsub_cmd [hunk 2]
```diff
@@ -31,17 +31,12 @@ import fake_ifdh
 class StoreGroupinEnvironment(argparse.Action):
     """Action to store the given group in the GROUP environment variable"""
 
-    def __call__(self, parser, namespace, values, option_string=None):
+    def __call__(self, parser: argparse.ArgumentParser, namespace:Any, values:dict[str,str], option_string=None:Union[None,str]):
```
Comment (shreyb) [inline bin/jobsub_cmd:34]: Same question as `condor_submit`

**Review by shreyb — CHANGES_REQUESTED** [review]
Looks great.  I just had a couple of questions.

File: bin/condor_submit (hunk 1 above)
Comment (marcmengel) [reply bin/condor_submit:53]: Mostly I didn't chase down what type it actually was... I can update that.

**Review by shreyb — APPROVED** [review]

**Pull request merged**